rust/agama-lib/src/software/model.rs

```diff
@@ -80,3 +80,23 @@ impl TryFrom<u32> for RegistrationRequirement {
         }
     }
 }
+
+/// Software resolvable type (package or pattern).
+#[derive(Deserialize, Serialize, strum::Display, utoipa::ToSchema)]
+#[strum(serialize_all = "camelCase")]
+#[serde(rename_all = "camelCase")]
+pub enum ResolvableType {
+    Package = 0,
+    Pattern = 1,
```

Comment:
btw future proof I think we will also need product resolvable type which can be used to set different product. But not related to this PR so just NP.